### ISSUE TITLE
tile entity state error?

### ISSUE THREAD

**Issue by rdelro** [Piepenguin1995/rfwindmill#17]
Note quite sure if the title is correct here, but I noticed after upgrading past basic to Resonant (with its similar rotor) My console spits this out: http://pastebin.com/N0ndiZGm (didn't mean to copy over the whole thing, but just read the first bit and you'll get it from there, i'm sure.)

**Comment by EmilyMansfield** [comment]
Thanks for the report, but I can't seem to reproduce this. Could you tell me the exact steps that caused this please?

**Comment by rdelro** [comment]
Well I started out with just one basic windmill, and that one did just fine (no rotor.) However once I tried breaking it with a pick, my client kicked me 'Internal Server Error' classic. So I tried a crescent hammer (Thermal Expansion) to pick it up...kicked again. Tried breaking it in creative...kicked. Tried a Terrain Smasher (Thermal, still.) finally breaks! However I cannot upgrade it. So I craft up some resonant tiers with resonant rotors and signalim. Added cables from Thermal Dynamics, separated the machines enough so they spin, attach cables under, attach rotors...wait...error.
TLDR: "Placed down basic, had trouble breaking it, used a machine to break it. Skipped all the way to resonant. Upgraded cables Leadstone > Redstone. Wait. Error." I'm sorry it's not a fancy story. Or perhaps anything that helps you. Other than the part where I couldn't break the blocks or perhaps upgrading the cables?

**Comment by EmilyMansfield** [comment]
Thank you, that's very helpful! I'll look into it again soon.

**Comment by dronf** [comment]
Probably a related issue, but I had 4 hardened windmills that were placed in the previous version, and worked fine.  After the update to .5, they no longer generated RF, and I would get booted with the Internal Server Error when trying to break/wrench them.  I could NOT attach a rotor to them, right clicking did nothing.  I tried breaking all the blocks around them, and noticed that they were generating a very small amount of power (i think 1.8/t at the most), even without a rotor.  I'll try the terrain smasher next.

**Comment by rdelro** [comment]
Oh, dronf! You may wanna try a BuildCraft wrench? As that's the only wrench that seems 100% compatible with no problems in many tech mods. I personally don't think I've ever had a problem with it. The Thermal Expansion 'hammers' tend to be slightly more buggy sadly. Just thought I'd share.

**Comment by dronf** [comment]
Same thing with BC wrench, "Internal Server Error".  I didn't yet have a chance to try the terrain smasher.

**Comment by rdelro** [comment]
Really? How odd! A straight vanilla BuildCraft wrench seems to always work. Well yeah try the terrain smasher for sure! Or any block breaker really. Cheers!

**Comment by EmilyMansfield** [comment]
This seems to be due to the tile entity failing to be registered by forge, which vesion of forge are you using?

**Comment by rdelro** [comment]
Forge version: 10.13.3.1355 if that helps! It's not the latest version but if I update, some mods break as well as my server...

**Comment by EmilyMansfield** [comment]
Thanks, I'll try running with that version and see if I get the same issue. In the meantime could you try the [latest developer build](https://github.com/Piepenguin1995/rfwindmill/releases/download/v0.5/rfwindmill-1.7.10-0.5a.jar) and see if that fixes the issue?

**Comment by rdelro** [comment]
Alright. Yes, of course! I'll test that right after I eat breakfast ^^;

**Comment by rdelro** [comment]
Alrighty http://i.imgur.com/AXqZ5tf.png I have installed the dev version onto the server and my clients. It works perfectly fine with all noted wrenches in the image. As well as listing bad wrenches. I didn't try every wrench (as I only use a select number of mods..) but I did use common wrenches as voted by the users on not only my server but many forum posts that commonly report using listed wrenches. In other words, it works! The bad wrenches give an IC2 api error, but that has nothing to do with your mod.

**Comment by EmilyMansfield** [comment]
Wonderful! It seems I'd accidentally set the entity registration for the rotors to be client side only, so it would crash if you were running a dedicated server, which is something I hadn't checked. Thanks for your help, I'll release the new version immediately.
